process/transaction/shardProcess.go

```diff
@@ -526,13 +532,43 @@ func (txProc *txProcessor) processRelayedTx(
 	return txProc.processUserTx(tx, userTx, adrSrc, tx.Value, tx.Nonce, txHash)
 }
 
-func (txProc *txProcessor) computeRelayedTxFees(tx *transaction.Transaction) (*big.Int, *big.Int) {
+func (txProc *txProcessor) computeRelayedTxFees(tx *transaction.Transaction) (*big.Int, *big.Int, uint64) {
 	relayerGasLimit := txProc.economicsFee.ComputeGasLimit(tx)
 	relayerFee := big.NewInt(0).Mul(big.NewInt(0).SetUint64(relayerGasLimit), big.NewInt(0).SetUint64(tx.GetGasPrice()))
 	totalFee := big.NewInt(0).Mul(big.NewInt(0).SetUint64(tx.GetGasLimit()), big.NewInt(0).SetUint64(tx.GetGasPrice()))
 	remainingFee := big.NewInt(0).Sub(totalFee, relayerFee)
 
-	return totalFee, remainingFee
+	return totalFee, remainingFee, tx.GasLimit - relayerGasLimit
```

Comment:
Could be here an overflow if tx.GasLimit - relayerGasLimit < 0  ? It is better to use int64 and check it outside

Reply:
no need as it is checked before in checkTxValidity

Reply:
ok

Reply:
ok